### PR TITLE
Change response structure for record summary

### LINKED ISSUE #3801: Return record summary data in json object instead of array
I have a request for a small change to the JSON schema of the API response that returns record summary data.

The `preview_data` property currently looks like this...

```json
{
  "2": [
    {
      "key": 1234,
      "summary": "Foo bar"
    }
  ]
}
```

Instead, I'd like it to look like:

```json
{
  "2": {
    "1234": "Foo bar"
  }
}
```

@Anish9901 and I [discussed](https://matrix.to/#/!UZILDSNKobkelUYwBp:matrix.mathesar.org/$AqJIMFvkiWfJqRNyqZeruuDKzjRRoN0tSIpnRvOPVWU?via=matrix.mathesar.org&via=matrix.org) this on Matrix.

### PR DESCRIPTION
**Pull request by Anish9901**
Fixes #3801

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by seancolsen — COMMENTED** [review]
This is returning results like:

```json
{
  "2": [
    {
      "1234": "Foo bar"
    },
    {
      "5678": "Baz bat"
    }
  ]
}
```

Instead, I want results like:

```json
{
  "2": {
    "1234": "Foo bar",
    "5678": "Baz bat"
  }
}
```

**Comment by Anish9901** [comment]
Ahh, should be now @seancolsen.

**Review by seancolsen — APPROVED** [review]
Thanks. This gives me what I need now 🙂

File: mathesar/rpc/records.py [hunk 1]
```diff
@@ -152,12 +140,13 @@ class RecordList(TypedDict):
...
         grouping: Information for displaying grouped records.
-        preview_data: Information for previewing foreign key values.
+        preview_data: Information for previewing foreign key values,
+            provides a map of foreign key to a text summary.
     """
     count: int
     results: list[dict]
     grouping: GroupingResponse
-    preview_data: dict[str, list[PreviewEntry]]
+    preview_data: dict[str, dict[Any, str]]
```
Comment (mathemancer) [inline mathesar/rpc/records.py:149]: The key will always be a string due to the nature of `jsonb_object_agg`. So, this should be:

```suggestion
    preview_data: dict[str, dict[str, str]]
```

@seancolsen To double-check, are you sure that mistyping won't cause a problem?

File: mathesar/rpc/records.py [hunk 2]
```diff
@@ -181,10 +170,11 @@ class RecordAdded(TypedDict):
 
     Attributes:
         results: An array of a single record objects (the one added).
-        preview_data: Information for previewing foreign key values.
+        preview_data: Information for previewing foreign key values,
+            provides a map of foreign key to a text summary.
     """
     results: list[dict]
-    preview_data: dict[str, list[PreviewEntry]]
+    preview_data: dict[str, dict[Any, str]]
```
Comment (mathemancer) [inline mathesar/rpc/records.py:177]: Same

```suggestion
    preview_data: dict[str, dict[str, str]]
```

File: mathesar/tests/rpc/test_records.py [hunk 3]
```diff
@@ -64,7 +64,7 @@ def mock_list_records(
                 {"id": 3, "count": 8, "results_eq": {"1": "lsfj", "2": 3422}}
             ]
         },
-        "preview_data": {"2": [{"key": 12345, "summary": "blkjdfslkj"}]},
+        "preview_data": {"2": {"12345": "blkjdfslkj"}},
```
Comment (mathemancer) [inline mathesar/tests/rpc/test_records.py:67]: The issue I mentioned above is visible here.

**Review by mathemancer — APPROVED** [review]
One small typing issue.

If it's not a problem for @seancolsen , please just fix the type annotation and merge. If it is, we need to rethink the response form since PostgreSQL can't do non-text JSON keys.

I'm going to approve (provisionally) since in the former case it's a small change and I don't want to hold things up.

**Comment by mathemancer** [comment]
@seancolsen I re-requested review from you to check that the typing issue I mentioned won't be a problem.

**Pull request merged**